### ISSUE TITLE
Typescript missing

### ISSUE THREAD

**Issue by sameerveda** [Somepub/minimal-web_template#1]
Love the idea but dont agree with choice of caddy server.
- I am someone who avoid having global dependecies, and it works well when you intend to share the project. but with caddy, if i work on a project with someone else, he'll also have to install caddy (additional step).
- Project requires `typescript`, but it is not listed in package,json (assuming user will have typescript installed in global is bad).
![image](https://user-images.githubusercontent.com/22796631/122665099-f89f9280-d1c2-11eb-8dc8-609e8049524b.png)

Overall great idea, i intend to use this with some [redom](https://redom.js.org/) projects

**Comment by Somepub** [comment]
Hello and thanks for the feedback! 

- `Typescript`: This was my fault, I will fix it, next time I should try to install it in a clean environment.
- `HTML`: I had this reworked, I just haven't committed this yet.
- `win32`: I never use(d) Windows or Mac for development, but I'll give it a try to support it.
- `degit`: never used it, so I'll give it a try.

Now the caddy part, as I mentioned in README, this template devServer is not "out of the box". But for developers, this is bad and needs extra steps. So I have been thinking for few days about how to solve this, my proposal would be:

- Add caddy binary, no need to download then, but I have to update the binary every time
- Add serve-http (or something else) as the DevServer as you mentioned in your repo

But let me explain why I didn't add serve-http or other js libs in the first place. I have been working on micro-frontends, I'm using Single-Spa to clue micro frontends and Lerna to manage them.
When I start all the micro-frontends with Webpack e.g `lerna exec --parallel -- npm run start` (20 separate repositories), it takes about ~15GB of ram and 100% CPU on all threads. So when I created this template I had in mind that I won't be using any JS libs for devServer. But I could be wrong and maybe serve-http (or other good JS devServer) is lightweight, I need to test it.

Also, do you know any kind of devServer that is not written in JS and come "out of the box" like Webpack and Rollup?

**Comment by sameerveda** [comment]
- It works fine on `win32`
- Include `serve-http` as dev server as `estrella` docs uses this as dev-server so finding example will be easy for user.
- `lerna + 20 repos` is a special case, that will rarely happen, so in that case developer can use specilize solution. and i agree with caddy. 

just missing css sourcemaps, otherwise an awesome template. Thanx

**Comment by MartinMuzatko** [comment]
I use caddy as lightweight reverse proxy. But all my containers run their own node server via express or http-serve etc.
For more sophisticated applications I make heavy use of traefik.

I would also prefer to take care of the server (dev or prod) myself or just use a server with watch included.

**Comment by Somepub** [comment]
Hello, again @sameerveda ,

I tried to replace Caddy with `serve-http`, but I ran into some troubles.
With node version >=14, I'm gettings an error `Unhandled exception: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js`.
I had the same error on your template.
How did it work for you? I'm using Debian 10

**Comment by sameerveda** [comment]
@Somepub which platform are you using ? i am using windows.
It seems node 14 + Linux has this issue. I dont have access to linux system, so i wont be much of help. can you try if if `https://www.npmjs.com/package/live-server` works on not.

**Comment by sameerveda** [comment]
@MartinMuzatko I agree. After compining the project, we can decide which server to use. But for fast prototype/dev i prefer something comes bundled with project template.

**Comment by sameerveda** [comment]
replace `fs.mkdir(outDir, { recursive: true }, console.log);` with `fs.mkdir(outDir, { recursive: true }, err => err && console.log(err));`

at start console prints a null which is confusing
